src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -158,6 +158,15 @@ ident = Ident . qname' . name
 set :: Expr
 set = ident "Set"
 
+-- | Creates a @data@ declaration with the given name, binding the list of type
+--   variables and defining the list of constructors.
+--
+--   > name [(α₁ : eₙ), …, {αₙ : eₙ}] [C₁, …, Cₘ]
+--   >   ↧
+--   > data name (α₁ : eₙ) … {αₙ : eₙ} : Set where
```

Comment:
To me it is not obvious what `(α₁ : eₙ) … {αₙ : eₙ}` should mean exactly. I guess that you want to express that the binders can be explicit or implicit. But the notation suggests that all but the last binder are explicit. I would just introduce meta variables `b₁ … bₙ` to express that the binders can be arbitrary.

Instead of `name` I'd suggest to use `D` since that is the meta variable we usually use for the names of data type declarations in the remaining documentation.